packages/x-charts/src/internals/shallowEqual.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Performs equality by iterating through keys on an object and returning false
+ * when any key has values which are not strictly equal between the arguments.
+ * Returns true when the values of all keys are strictly equal.
+ *
+ * Source: https://github.com/facebook/react/blob/c2a196174763e0b4f16ed1c512ed4442b062395e/packages/shared/shallowEqual.js#L18
+ */
+export function shallowEqual(objA: unknown, objB: unknown): boolean {
```

Comment:
Did you check if our current implementation was useful enough for the case? Doesn't seem necessary to have two funcs that do the same thing.

Reply:
Yeah. I wanted to use the same algorithm as React for comparing props to ensure it works as expected. Also, `fastObjectShallowCompare` does not have the same behavior. For example:

1. `shallowEqual(NaN, NaN) === true`, but `fastObjectShallowCompare(NaN, NaN) === false`
2. `shallowEqual(+0, -0) === false`, but `fastObjectShallowCompare(+0, -0) === true`

Also, `shallowEqual` uses `Object.keys` while `fastObjectShallowCompare` uses `for...in` for obtaining the keys of an object, and those have different behaviors. 

So, I think it's better to use `shallowEqual` because `fastObjectShallowCompare` isn't equivalent and the goal is to mirror React's prop comparison algorithm.